service/matching/task_queue_manager.go

```diff
@@ -799,9 +803,17 @@ func (c *taskQueueManagerImpl) RedirectToVersionedQueueForAdd(ctx context.Contex
 	// Have to look up versioning data.
 	userData, userDataChanged, err := c.GetUserData()
 	if err != nil {
-		if errors.Is(err, errUserDataDisabled) && buildId == "" {
+		if errors.Is(err, errUserDataDisabled) {
 			// When user data disabled, send "default" tasks to unversioned queue.
-			return c.taskQueueID, userDataChanged, nil
+			if buildId == "" {
+				return c.taskQueueID, userDataChanged, nil
+			}
+			// Send versioned sticky back to regular queue so they can go in the dlq.
+			if c.kind == enumspb.TASK_QUEUE_KIND_STICKY {
+				return nil, nil, serviceerrors.NewStickyWorkerUnavailable()
```

Comment:
Can you test this?

Reply:
The integration test exercises this path. I know because I had to add it before the test would pass :)